sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala

```diff
@@ -75,6 +107,45 @@ case class StreamingExecutionRelation(
   )
 }
 
+/**
+ * Used to link a streaming [[org.apache.spark.sql.sources.v2.DataSourceV2]] of data into a
+ * [[org.apache.spark.sql.catalyst.plans.logical.LogicalPlan]].
+ */
+case class StreamingExecutionRelationV2(
```

Comment:
I tried having them merged, and they end up just looking like two different classes glued together.

I also think keeping them separate makes the migration cleaner, since we never have to wonder what the "real" type of a node is.